README.md

```diff
@@ -233,32 +258,84 @@ Following are about the toggles:
 
+- [`packages/convenient_test_manager`](packages/convenient_test_manager) is a flutter GUI app that allows running and inspecting individual tests.
+  Use this for development and debugging.
+- [`packages/convenient_test_manager_dart`](packages/convenient_test_manager_dart) is a dart CLI app that runs all tests and terminates with a corresponding exit code.
+  Use this for CI and automated testing. See [CI / headless mode](#ci--headless-mode)
+
 #### Method 1: Download precompiled binary
 
-There are precompiled binaries for each commit, at the "artifacts" section in [the CI page](https://github.com/fzyzcjy/flutter_convenient_test/actions/workflows/ci.yaml). For example, download `manager_macos` artifact, unzip it, and open it.
+See [Releases](https://github.com/fzyzcjy/flutter_convenient_test/releases) for binaries.
+
+Additionally, there are precompiled binaries for each commit, at the "artifacts" section in [the CI page](https://github.com/fzyzcjy/flutter_convenient_test/actions/workflows/ci.yaml)
+(download only available for logged in GitHub users).
+For example, go to the most recent _CI_ run, download `manager_gui_macos` artifact, unzip it, and open it.
+
+#### Method 2: `flutter run` / `flutter build`
+
+Check the requirements for developing flutter desktop apps for
+[macOS](https://docs.flutter.dev/platform-integration/macos/install-macos/install-macos-from-ios),
+[Linux](https://docs.flutter.dev/platform-integration/linux/install-linux/install-linux-from-android),
+and [Windows](https://docs.flutter.dev/platform-integration/windows/install-windows/install-windows-from-android).
+
+```sh
+git clone https://github.com/fzyzcjy/flutter_convenient_test.git
+cd flutter_convenient_test/packages/convenient_test_manager
 
-Currently only MacOS and Linux has CI for this, and I am willing to accept a PR for Windows.
+# compile executable. platform = linux | macos | windows
+flutter build <platform>
+# or just run directly
+flutter run -d <platform>
+```
+
+##### For CLI manager:
+
+```sh
+cd flutter_convenient_test/packages/convenient_test_manager
+
+dart compile exe bin/convenient_test_manager_dart.dart
+# or just run directly
+dart run
+```
+
+#### Method 3: `dart pub global` (only CLI manager)
 
-#### Method 2: `flutter run` / `flutter profile` / `flutter build`
+Set up once (re-run to update):
 
-`packages/convenient_test_manager` is nothing but a *normal* Flutter Windows/MacOS/Linux app, so run it following the Flutter official doc.
+```sh
+dart pub global activate --source git https://github.com/fzyzcjy/flutter_convenient_test.git --git-path packages/convenient_test_manager_dart
+```
+
+> [!TIP]
+> Add `--git-ref v1.x.x` to `dart pub global activate` to check out a specific [release](https://github.com/fzyzcjy/flutter_convenient_test/releases).
+
+Run using:
 
-Surely, it can also be run via `flutter profile` mode to speed up. Or use `flutter build` and the standard approaches to generate a release version binary and use it.
```

Comment:
(similarly maybe keep this)

Reply:
I have taken out the reference to `profile` on purpose (on an aside, the command is `flutter run --profile` in case we do keep it in) because it has no clear benefit over the other options _in this case_.

Yes, the app -- once build -- may be faster than `debug`, but so does `--release` (technically even faster) and profile has no benefit over release in this use-case (also takes the same time to build -- i tested). So if you want better performance (i argue this does not matter in most cases when people use the GUI) then just _build_ the executable.

"Then why not recommend the `--release` flag for `flutter run`" - because this *does* slow down the build significantly (the same as `--profile`). So `flutter run` (implicit debug) is the "quick and dirty" option for devs who just want to get to testing as quick as possible.

I thought about explaining all this in the readme, but this takes up too much space and may make the getting started section unnecessarily complicated. So i simply left it out.

IMO there are two things that may provide a benefit to user while keeping it reasonably concise:

- link to [build modes in the flutter docs](https://docs.flutter.dev/testing/build-modes). but this may waste the time of newbies who simply need to run a program to orchestrate tests
- add a _very_ concise version of the above TED talk as an `> [!TIP]` info box

but i believe it is acceptable to leave it as is

Reply:
I see and totally agree!